src/hooks/useInitialLoadingStatus.ts

```diff
@@ -0,0 +1,25 @@
+import { useFeatureFlag } from './useFeatureFlags';
+
+type InitialLoadingStatus =
+    | 'READY'
+    | 'LOADING_DEFAULT_DRIVES'
+    | 'SYNCING_DRIVES';
+
+export const useInitialLoadingStatus = () => {
```

Comment:
I'd add an optional argument to set a drive id here.
So when the url has a file in the path, we just want to wait for that specific drive, that the document belongs to, to finish loading.

Reply:
added here: [5a8867c](https://github.com/powerhouse-inc/connect/pull/482/commits/5a8867c1347c6808afc0a63c9cea33bba5f718a0)